Expand ConnectionStart doc. (#1628)
microsoft/msquic: docs/api/ConnectionStart.md

# Remarks

`ConnectionStart` initiates the connection from the client application. A server application doesn't start its side of the connection until it returns `QUIC_STATUS_SUCCESS` from the `QUIC_LISTENER_EVENT_NEW_CONNECTION` event.

> Since 0-RTT data is opportunistically sent during the connection handshake, it must be queued for send **BEFORE** calling `ConnectionStart` otherwise it will be sent after the handshake completes.

Reviewer: ```suggestion
Since 0-RTT data is opportunistically sent during the connection handshake, it should be queued for send **BEFORE** calling `ConnectionStart` otherwise it may be sent after the handshake.
```
It's a race condition. It's possible it might send together, but likely won't.
